File: eolimp/page17/q1674/Main.c
#include <stdio.h>
#define max(a, b) a > b ? a : b
#define min(a, b) a < b ? a : b

// There is a weird bug on the judge
// throws runtime in java
int main() {
    int a, b, c, t, n;
    scanf("%d %d %d\n%d", &a, &b, &c, &n);
    while(n-- > 0) {
        scanf("%d", &t);
        c = t < 0 ? max(c + t, a) : min(c + t, b);
    }
    printf("%d\n", c);
    return 0;
}

File: eolimp/page17/q1674/Main.java
package page17.q1674;

import java.util.Scanner;

public class Main {
  public static void main(String[] args) {
    Scanner in = new Scanner(System.in);
    int a = in.nextInt(), b = in.nextInt(),
        s = in.nextInt(), n = in.nextInt(), t;

    while(n-- > 0) {
      t = in.nextInt();
      s = t > 0 ? Math.min(s + t, b) : Math.max(s + t, a);
    }
    System.out.println(s);
  }
}
